[Industries] Side Nav 컴포넌트와 산업 특징, 산업 관련 제품 컴포넌트 초안

ryudg opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- Industries 페이지의 기능 및 레이아웃 (header, main, nav, footer 등 공통 section 을 제외) 을 구현하였습니다.

## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

- 테블릿, 데스크탑, 모바일 반응형 레이아웃 구현
- SNB 컴포넌트로 현재 페이지의 해당 섹션으로 이동
- 해당 산업군의 더미 데이터(Json)를 이용해 산업군 특징 컴포넌트 구현
- 해당 산업군에 관련된 제품 및 특징을 산업군과 솔루션 더미 데이터을 이용해 관계를 형성해 리스트 렌더링


https://github.com/ReOpenSG/ReOpenSG/assets/103430498/e70dd5fa-a694-42b9-af1a-6c17fdbc55f1


https://github.com/ReOpenSG/ReOpenSG/assets/103430498/59e302bd-8694-4a79-a326-b47bbe6afb72




## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

- 없음

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요
 
- 없음

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

- 없음

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

ryudg committed:
➕ [Add] : Snb Scrollspy 이벤트를 위한 패키지 추가
- reac-intersection-observer 패키지 추가

ryudg committed:
➕ [Add] : industriesData, solutionsData 추가
- industriesData.js : 산업별 데이터 추가
  - 산업별 데이터는 solutionsData.json에 있는 솔루션들을 산업별로 분류하여 추가
- solutionsData.json : 솔루션별 데이터 추가

ryudg committed:
✨ [Feat] : Char 컴포넌트 작성
- 해당 산업군에 대한 산업 특징을 설명하는 컴포넌트 작성

Github issue #15

ryudg committed:
✨ [Feat] : Product 컴포넌트 작성
- 산업군에 관련된 제픔을 설명하는 컴포넌트 작성
  - solutionData.json에 작성되어있는 솔루션의 산업군을,
    industriesData.js의 각 산업군별 solutions 배열에 매칭시켜 추가 후
    industriesData.js에 작성되어있는 산업군의 이름과 해당 주소를 비교하여 일치하는 솔루션을 렌더링

Github issue #16

ryudg committed:
✨ [Feat] : Snb 컴포넌트 작성
- react-intersction-observer 라이브러리를 사용하여 스크롤에 따라 Snb 메뉴가 활성화 되도록 구현
  - 해당하는 섹션에 진입하면 Snb 메뉴가 활성화 되도록 구현

Github issue #17

ryudg committed:
✨ [Feat] : Industries 컴포넌트 작성
- Char, Product, Snb 컴포넌트를 사용해 컴포넌트 구현
  - 반응형 디자인 적용

Github issue #14

ryudg committed:
✨ [Feat] : 관련 제품 링크 추가
- Solution 별 링크 추가
  - 해당 Solution 페이지로 이동

Github issue #16

ryudg committed:
🌈 [Design] : Industries 컴포넌트의 tailwindcss modulecss로 변경
- tailwindcss modulecss로 변경
- 절대 경로 설정
- tailwindcss 변수 추가

Github issue #14

ryudg committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

ryudg committed:
🐛 [Fix] : ul 요소를 dl 요소로 변경
- 기존의 ul 요소를 dl 요소로 변경
  - ul 요소는 목록을 나타내는 요소이지만, dl 요소는 정의 목록을 나타내는 요소이기 때문에 더 적합하다고 판단

Github issue #16

kelvin-chihyun reviewed (approved):
P3: 확인했고, 이상없어 보입니다.

ingbinsee commented on src/components/Industries/Snb.jsx:
P3: 관련 제품으로 이동 시 SNB 색상 변경이 되지 않습니다. 해당 부분 확인 후 수정 요청드립니다.

ryudg replied:
`industries/port`의 산업군 - 항만 페이지에서 SNB 색상이 변경되지 않는 문제는 footer 등 다른 컨텐츠가 추가되면 해결 가능합니다

ingbinsee reviewed (approved):
아래 파일에 의견 남겨두었습니다. 확인 부탁드립니다.

Snb.jsx (P1 review)

ingbinsee reviewed (approved):
아래 파일에 의견 남겨두었습니다. 확인 부탁드립니다.

Snb.jsx (P3 review) ```P1에서 P3으로 변경```

dani-day commented on src/components/Industries/Product.jsx:
```diff
@@ -0,0 +1,39 @@
+                <img src="http://via.placeholder.com/200x250" alt="제픔 사진" />
+              </div>
+              <dl className={styles.list}>
+                <dt className={styles.itemName}>{item[0]}</dt>
+                <dd className={styles.itemSubName}>{item[1].제품명}</dd>
+                <dd className={styles.itemDesc}>{item[1].제품설명}</dd>
```
P2: 내용상 제품명이 약어와 full name 2 개로 구성되어있는 경우 제품명에는 모두 dt 태그를 사용하고 설명에만 dd 태그를 주는게 더 적절하지 않나 의견드립니다 :)

[Multiple terms, single description](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl#multiple_terms_single_description)

dani-day reviewed (approved):
코드가 간결하고 보기 쉽게 정돈되어있어 리뷰하기 편했습니다 😄 
- Products.jsx 에 P2 의견 1개 남겨두었습니다.

ryudg committed:
🐛 [Fix] : 기존 `dd` 태그 `dt` 태그로 변경
- 기존 제품명이 약어와 full name으로 구성되어있는 경우,
  full name을 `dd` 태그로 사용했는데, `dt` 태그로 변경

Github issue #14

ryudg committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화
